DllImportGenerator/DllImportGenerator/ManualTypeMarshallingHelper.cs

```diff
@@ -23,17 +27,34 @@ public static bool HasToManagedMethod(ITypeSymbol nativeType, ITypeSymbol manage
                         && !m.IsStatic);
         }
 
-        public static bool IsManagedToNativeConstructor(IMethodSymbol ctor, ITypeSymbol managedType)
+        public static bool IsManagedToNativeConstructor(
+            IMethodSymbol ctor,
+            ITypeSymbol managedType,
+            bool isCollectionMarshaller)
```

Comment:
For future proofing, it might be worth making this an enum. That way if we ever introduce another marshaller model extension, we can easily extend these APIs cleanly.